src/macro-context.js

```diff
@@ -142,7 +142,8 @@ ctx :: {
 */
 export default class MacroContext {
   constructor(enf, name, context, useScope, introducedScope) {
-    const { term, rest, prev, done} = this[symEnf] = enf;
+    const { term, rest, prev, done } = enf;
+    this[symEnf] = new Enforester(rest, prev, enf.context);
```

Comment:
@disnet: This avoids mutating the enforester that was passed in. If you want I can do what you suggested and instantiate a new one each time `prev` is called.